Source/Plugins/Core/com.equella.core/js/tsconfig.json

```
@@ -3,7 +3,7 @@
     "outDir": "target/ts",
     "module": "CommonJS",
     "target": "es5",
-    "lib": ["es6", "dom"],
+    "lib": ["es6", "dom", "es2018.promise"],
```

Since TS is going to compile this down to ES5 would there be any issues setting this to support all features up to ES2020?

```suggestion
    "lib": ["es2020", "dom"],
```

---

Heya @ChristianMurphy 

Just to be sure to be sure, I wonder if we should do that in a separate stand-alone PR? Just in case it triggers any other changes in the transpilation. Thoughts?

---

Oh whoops. I just realised that @PenghaiZhang has brought in `es2018.promise`.

I guess in that case, maybe here is as good as any to bring in the whole thing and see what happens. It's a big enough change anyway.

---

> Just to be sure to be sure, I wonder if we should do that in a separate stand-alone PR? Just in case it triggers any other changes in the transpilation. Thoughts?

I'd be okay either way.

> I guess in that case, maybe here is as good as any to bring in the whole thing and see what happens. It's a big enough change anyway.

My thoughts precisely

---

Do we have handling for legacy browsers?  E.g. message stating browser is not supported

---

this leaves the target for the output the same so browsers should be unaffected.

The closest I've seen oEQ having to an official browser matrix is https://github.com/openequella/openequella.github.io/blob/7001aad5156fd20a910acabb6e8314b239a30bb3/guides/FileManagerGuide.md#not-supported-without-reconfiguration-if-at-all